libnavui-androidauto/src/main/java/com/mapbox/androidauto/car/feedback/ui/CarGridFeedbackScreen.kt

```diff
@@ -65,12 +67,15 @@ class CarGridFeedbackScreen constructor(
     private fun buildItemList(options: List<CarFeedbackOption>): ItemList {
         val itemListBuilder = ItemList.Builder()
         for (option in options) {
-            val gridItem = GridItem.Builder()
-                .setTitle(option.title)
-                .setImage(option.icon, GridItem.IMAGE_TYPE_ICON)
-                .setOnClickListener { selectOption(option) }
-                .build()
-            itemListBuilder.addItem(gridItem)
+            val itemBuilder = GridItem.Builder().setTitle(option.title)
+            val image = iconDownloader.getOrDownload(option.icon)
```

Comment:
Curious what happens when downloading the 6 images like in the PR screen recording.

How many templates are generated? Does it create a bunch of them?

Wondering  because the screen invalidator, could be a Flow where there is a debounce or something

Reply:
I see that `onGetTemplate` is called multiple times, almost once for each download. Can that be a problem?
From [`GridTemplate docs`](https://developer.android.com/reference/androidx/car/app/model/GridTemplate#template-restrictions) if screen title is not changing, number of grid items and the title of each item is not changing, then each template is considered an update of a previous one, so we don't risk violating template restrictions. Also I tried changing images of the items infinitely every second and it worked without any problems.

Reply:
Looks like `invalidate` has a built in debouncer, because I see that after downloading 5 images there are 5 `invalidate` calls, but `onGetTemplate` is usually called 3 times, sometimes 2 or 4. Thus we probably don't have to add additional optimizations.